ci_framework/roles/devscripts/tasks/sub_tasks/42_add_bmh.yml

```diff
@@ -45,27 +45,27 @@
       loop_control:
         loop_var: node
 
-- name: Ensure the bmh artifacts exists
-  ansible.builtin.file:
-    path: "{{ cifmw_devscripts_artifacts_dir }}/bmh"
-    state: "directory"
-    owner: "{{ cifmw_devscripts_user }}"
-    group: "{{ cifmw_devscripts_user }}"
-    mode: "0755"
+    - name: Ensure the bmh artifacts exists
+      ansible.builtin.file:
+        path: "{{ cifmw_devscripts_artifacts_dir }}/bmh"
+        state: "directory"
+        owner: "{{ cifmw_devscripts_user }}"
+        group: "{{ cifmw_devscripts_user }}"
+        mode: "0755"
 
-- name: Create baremetal host object
-  vars:
-    node: "{{ item }}"
-  ansible.builtin.template:
-    src: templates/bmh.yaml.j2
-    dest: "{{ cifmw_devscripts_artifacts_dir }}/bmh/{{ item.name }}.yaml"
-    owner: "{{ cifmw_devscripts_user }}"
-    group: "{{ cifmw_devscripts_user }}"
-    mode: "0644"
-  loop: "{{ cifmw_devscripts_osp_compute_nodes }}"
+    - name: Create baremetal host object
+      vars:
+        node: "{{ item }}"
+      ansible.builtin.template:
+        src: templates/bmh.yaml.j2
+        dest: "{{ cifmw_devscripts_artifacts_dir }}/bmh/{{ item.name }}.yaml"
+        owner: "{{ cifmw_devscripts_user }}"
+        group: "{{ cifmw_devscripts_user }}"
+        mode: "0644"
+      loop: "{{ cifmw_devscripts_osp_compute_nodes }}"
 
-- name: Apply the baremetal host definitions
-  environment:
-    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
-  ansible.builtin.command:
-    cmd: "oc apply -f {{ cifmw_devscripts_artifacts_dir }}/bmh/"
+    - name: Apply the baremetal host definitions
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+      ansible.builtin.command:
+        cmd: "oc apply -f {{ cifmw_devscripts_artifacts_dir }}/bmh/"
```

Comment:
This should not come inside the block... it is supposed to be outside

`cifmw_devscripts_osp_compute_nodes` is either provided by the user as a param or is set here when `num_extra_nodes` is set

what is missing code be a block that checks `cifmw_devscripts_osp_compute_nodes`

Reply:
If we move it into the block then it would break when physical computes are being provided.